src/framework/pool_hash.c

```diff
@@ -1,21 +1,25 @@
 #include <stdlib.h>
 
-#ifdef UNDERSCORE
-#define pool_hash pool_hash_
-#else
-#ifdef DOUBLEUNDERSCORE
-#define pool_hash pool_hash__
-#endif
-#endif
+/*
+ use iso_c_binding, only : c_int, c_char
 
-void pool_hash(int* hash, char* key, int* len)
+ interface
+   subroutine pool_hash(hash, key, len) bind(c)
+      use iso_c_binding, only : c_int, c_char
+      integer (c_int), intent(inout) :: hash
+      character (c_char), intent(in) :: key
+   end subroutine pool_hash
+ end interface
+*/
+
+void c_pool_hash(int* hash, char* key)
 {
 	int i;
 	unsigned int whash;
 
 	whash = 0;
 
-	for (i=0; i<(*len); i++) {
+	for (i=0; i < strlen(key); i++) {
```

Comment:
If we are guaranteed that the `key` that is passed in is a null-terminated string, then we should be able to avoid an extra O(n) operation here by looping until we hit a `\0` character.